src/ZiggyCreatures.FusionCache.Backplane.NATS/NatsBackplane.cs

```diff
@@ -0,0 +1,142 @@
+﻿using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+using NATS.Client.Core;
+
+namespace ZiggyCreatures.Caching.Fusion.Backplane.NATS;
+
+/// <summary>
+/// A NATS based implementation of a FusionCache backplane.
+/// </summary>
+public partial class NatsBackplane
+	: IFusionCacheBackplane
+{
+	private BackplaneSubscriptionOptions? _subscriptionOptions;
+	private readonly ILogger? _logger;
+	private INatsConnection _connection;
+	private string _channelName = "";
+	private Func<BackplaneMessage, ValueTask>? _incomingMessageHandlerAsync;
+	private INatsSub<NatsMemoryOwner<byte>>? _subscription;
+
+	/// <summary>
+	/// Initializes a new instance of the NatsBackplane class.
+	/// </summary>
+	/// <param name="natsConnection">The NATS connection instance to use.</param>
+	/// <param name="logger">The <see cref="ILogger{TCategoryName}"/> instance to use. If null, logging will be completely disabled.</param>
+	public NatsBackplane(INatsConnection? natsConnection, ILogger<NatsBackplane>? logger = null)
+	{
+		_connection = natsConnection ?? throw new ArgumentNullException(nameof(natsConnection));
+
+		// LOGGING
+		if (logger is NullLogger<NatsBackplane>)
+		{
+			// IGNORE NULL LOGGER (FOR BETTER PERF)
+			_logger = null;
+		}
+		else
+		{
+			_logger = logger;
+		}
+	}
+
+	/// <inheritdoc/>
+	public async ValueTask SubscribeAsync(BackplaneSubscriptionOptions subscriptionOptions)
+	{
+		if (subscriptionOptions is null)
+			throw new ArgumentNullException(nameof(subscriptionOptions));
+
+		if (subscriptionOptions.ChannelName is null)
+			throw new NullReferenceException("The BackplaneSubscriptionOptions.ChannelName cannot be null");
+
+		if (subscriptionOptions.IncomingMessageHandler is null)
+			throw new NullReferenceException("The BackplaneSubscriptionOptions.IncomingMessageHandler cannot be null");
+
+		if (subscriptionOptions.ConnectHandler is null)
+			throw new NullReferenceException("The BackplaneSubscriptionOptions.ConnectHandler cannot be null");
+
+		if (subscriptionOptions.IncomingMessageHandlerAsync is null)
+			throw new NullReferenceException("The BackplaneSubscriptionOptions.IncomingMessageHandlerAsync cannot be null");
+
+		if (subscriptionOptions.ConnectHandlerAsync is null)
+			throw new NullReferenceException("The BackplaneSubscriptionOptions.ConnectHandlerAsync cannot be null");
+
+		_subscriptionOptions = subscriptionOptions;
+
+		_channelName = _subscriptionOptions.ChannelName;
+		if (string.IsNullOrEmpty(_channelName))
+			throw new NullReferenceException("The backplane channel name must have a value");
+
+		_incomingMessageHandlerAsync = _subscriptionOptions.IncomingMessageHandlerAsync;
+		_subscription = await _connection.SubscribeCoreAsync<NatsMemoryOwner<byte>>(_channelName);
+		_ = Task.Run(async () =>
+		{
+			while (await _subscription.Msgs.WaitToReadAsync().ConfigureAwait(false))
```

Comment:
I'm not sure but should we use a CancellationTokenSource+CancellationToken here to signal on unsubscribing?
As well should we handle any exceptions in these while loops?